libnopegl/src/node_renderother.c

```diff
@@ -401,6 +430,60 @@ static const struct node_param renderhistogram_params[] = {
 };
 #undef OFFSET
 
+#define NOISE_TYPE_BLOCKY 0
+#define NOISE_TYPE_PERLIN 1
+
+const struct param_choices noise_type_choices = {
+    .name = "noise_type",
+    .consts = {
+        {"blocky",    NOISE_TYPE_BLOCKY,  .desc=NGLI_DOCSTRING("blocky noise")},
+        {"perlin",    NOISE_TYPE_PERLIN,  .desc=NGLI_DOCSTRING("perlin noise")},
+        {NULL}
+    }
+};
+
+#define OFFSET(x) offsetof(struct rendernoise_opts, x)
+static const struct node_param rendernoise_params[] = {
+    {"type",        NGLI_PARAM_TYPE_SELECT, OFFSET(type),
+                    .choices=&noise_type_choices,
+                    .desc=NGLI_DOCSTRING("noise type")},
+    {"frequency",   NGLI_PARAM_TYPE_F32, OFFSET(frequency_node), {.f32=1.f},
+                    .flags=NGLI_PARAM_FLAG_ALLOW_NODE,
+                    .desc=NGLI_DOCSTRING("oscillation per second")},
+    {"amplitude",   NGLI_PARAM_TYPE_F32, OFFSET(amplitude_node), {.f32=1.f},
+                    .flags=NGLI_PARAM_FLAG_ALLOW_NODE,
+                    .desc=NGLI_DOCSTRING("by how much it oscillates")},
+    {"octaves",     NGLI_PARAM_TYPE_U32, OFFSET(octaves_node), {.u32=3},
+                    .flags=NGLI_PARAM_FLAG_ALLOW_NODE,
+                    .desc=NGLI_DOCSTRING("number of accumulated noise layers (controls the level of details)")},
+    {"lacunarity",  NGLI_PARAM_TYPE_F32, OFFSET(lacunarity_node), {.f32=2.f},
+                    .flags=NGLI_PARAM_FLAG_ALLOW_NODE,
+                    .desc=NGLI_DOCSTRING("frequency multiplier per octave")},
+    {"gain",        NGLI_PARAM_TYPE_F32, OFFSET(gain_node), {.f32=0.5f},
+                    .flags=NGLI_PARAM_FLAG_ALLOW_NODE,
+                    .desc=NGLI_DOCSTRING("amplitude multiplier per octave (also known as persistence)")},
+    {"seed",        NGLI_PARAM_TYPE_U32, OFFSET(seed_node), {.u32=0},
+                    .flags=NGLI_PARAM_FLAG_ALLOW_NODE,
+                    .desc=NGLI_DOCSTRING("random base seed")},
+    {"scale",       NGLI_PARAM_TYPE_VEC2, OFFSET(scale_node), {.vec = {32.f, 32.f}},
+                    .flags=NGLI_PARAM_FLAG_ALLOW_NODE,
+                    .desc=NGLI_DOCSTRING("size of the grid in lattice units")},
+    {"evolution",   NGLI_PARAM_TYPE_F32, OFFSET(evolution_node), {.f32 = 0.0},
+                    .flags=NGLI_PARAM_FLAG_ALLOW_NODE,
+                    .desc=NGLI_DOCSTRING("evolution of the 3rd non-spatial dimension, time if unspecified")},
```

Comment:
I thought we agreed on switching to a node-only parameter so that it's animated by default, or we don't want that anymore?